src/native/libs/System.Security.Cryptography.Native/opensslshim.c

```diff
@@ -127,6 +127,17 @@ static void OpenLibraryOnce(void)
     {
         DlOpen(MAKELIB("111"));
     }
+    // FreeBSD uses this for OpenSSL3 from ports. OpenSSL3.1 would be "13"
+    // Ports version of OpenSSL is used over base where possible
+    if (libssl == NULL)
+    {
+        DlOpen(MAKELIB("12"));
```

Comment:
.NET prefers OpenSSL 3 over OpenSSL 1.1.x. As the code is currently structured, it appears that 1.1 will be loaded instead of 3.0 on FreeBSD if both are present on the system.

I think it would make sense to move these new checks above the other FreeBSD OpenSSL 1.1 checks.